Manatee.Json/Schema/SchemaValidationContext.cs

```diff
@@ -104,12 +117,21 @@ public class SchemaValidationContext
 			RelativeLocation = relativeLocation;
 			InstanceLocation = instanceLocation;
 			LocalRegistry = new JsonSchemaRegistry();
+
+			ShouldTrackEvaluatedPropertyNames = shouldTrackEvaluatedPropertyNames ?? root.Any(k => k is UnevaluatedPropertiesKeyword);
```

Comment:
I don't think that we want the caller to be able to completely override whether the evaluated properties should be tracked.  If the schema *does* contain an `unevaluatedProperties` keyword, and they pass false, then it won't work right.

I think the right way to do this is default to false, and `||` with the keyword lookup.

Reply:
That makes sense. I made the setter on the properties public, because you mentioned people are encouraged to add their own keywords, and I wanted them to be able to enable the tracking if they needed it for their own keywords, but maybe instead of making the setters public, I should add "Enable" methods so that they can turn the tracking on if they need, but can't turn it off if it's needed for `unevaluatedProperties`. 

Thoughts?

Reply:
Welp, the library is targeting netstandard2.0 which doesn't support default implementations, but I still feel like letting keywords specify that they need the tracking is a better move. 

What about 2 empty interfaces, that keywords can implement to specify they need the tracking?